artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java

```diff
@@ -154,12 +154,16 @@ public int sendMessage(MessageReference ref,
             encoder.encode(buffer, bodySize);
             encoder.close();
          } else {
-            buffer = coreMessage.getReadOnlyBodyBuffer();
+            if (Boolean.TRUE.equals(serverMessage.getBooleanProperty(Message.HDR_LARGE_COMPRESSED))) {
+               buffer = coreMessage.getBodyBuffer();
```

Comment:
You should always use the readOnlyBodyBuffer...

Why did you need the writeable body buffer here? it's dangerous!

Reply:
this is because we need to decompress the content before send to clients. If readonly we can't write into it.